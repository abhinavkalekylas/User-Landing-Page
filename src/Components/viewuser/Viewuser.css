h2, p, h6,.title {
    color:black;
    margin-top: 1rem;
}

.modal-backdrop.show {
    opacity: 0.11;
}

p {
    font-size: 1.5rem;
}

.modal_container {
    font-family: Arial, Helvetica, sans-serif;
}