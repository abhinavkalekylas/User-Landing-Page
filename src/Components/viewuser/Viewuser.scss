$text-color: black;

.viewuser_container {
    .user_data {
        color: $text-color;
        margin-top: 1rem;
        font-size: 1.5rem;
        font-family: Arial, Helvetica, sans-serif;
    }
}
