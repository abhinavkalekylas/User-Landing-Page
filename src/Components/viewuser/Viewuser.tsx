import { Modal, Button } from "react-bootstrap";
import "./Viewuser.css";
import { User } from "../../modalfunction/Modal";

const Viewuser = ({
  userData,
  viewUserModal,
  closeModal,
}: {
  userData: User;
  viewUserModal: boolean;
  closeModal: () => void;
}) => {
  return (
    <Modal className="modal_container" show={viewUserModal} onHide={closeModal}>
      <Modal.Header closeButton>
        <Modal.Title className="title">User Details</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <p className="user_data">Name : {userData.name}</p>
        <p className="user_data">Email : {userData.email}</p>
        <p className="user_data">Gender : {userData.gender}</p>
        <p className="user_data">Current Status : {userData.status}</p>
      </Modal.Body>

      <Modal.Footer>
        <Button className="close_btn" color="primary" onClick={closeModal}>
          Close
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

export default Viewuser;
