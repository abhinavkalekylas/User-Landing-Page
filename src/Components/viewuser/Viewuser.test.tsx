import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import {
  GenderType,
  StatusType,
  User,
} from "../../helperfunction/helperfuntion";
import Viewuser from "./Viewuser";

it("renders correctly Viewuser component", () => {
  const userData: User = {
    id: 0,
    name: "shivam sharma",
    email: "shivam@gmail.com",
    gender: GenderType.MALE,
    status: StatusType.ACTIVE,
  };

  const props = {
    userData,
    showModal: true,
    handleClose: jest.fn(),
  };
  const wrapper = shallow(<Viewuser {...props} showModal={false} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

// Todo Close button
fit("test close button", () => {
  // const mockCallBack = jest.fn();
  // wrapper.find(".close_btn").simulate("click");
  // expect(mockCallBack.mock.calls.length).toEqual(1);
});
