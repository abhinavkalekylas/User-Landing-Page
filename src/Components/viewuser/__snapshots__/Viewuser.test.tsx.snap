// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Viewuser component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[MockFunction]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      className="title"
    >
      User Details
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <p>
      Name : 
      shivam sharma
    </p>
    <p>
      Email : 
      shivam@gmail.com
    </p>
    <p>
      Gender : 
      male
    </p>
    <p>
      Current Status : 
      active
    </p>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      className=".close_btn"
      color="primary"
      disabled={false}
      onClick={[MockFunction]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
